Index: JSP/src/main/java/com/nixsolutions/bondarenko/study/jsp/servlets/AdminServlet.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- JSP/src/main/java/com/nixsolutions/bondarenko/study/jsp/servlets/AdminServlet.java	(date 1480521789000)
+++ JSP/src/main/java/com/nixsolutions/bondarenko/study/jsp/servlets/AdminServlet.java	(revision )
@@ -19,6 +19,7 @@
     private static final String ERROR_NOT_UNIQUE_EMAIL = "Error! This email is already attached to another user";
     private static final String MESSAGE_USER_CREATED = "New user has been created";
     private static final String MESSAGE_USER_UPDATED = "User has been updated";
+    private static final String ERROR_WRONG_PASSWORD = "Old password is incorrect";
 
     public static final String ACTION_DELETE_USER = "delete";
     public static final String ACTION_CREATE_USER = "create_user";
@@ -184,9 +185,10 @@
     private boolean processUser(User user, String action, HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
         try {
             User userByEmail = jdbcUserDao.findByEmail(user.getEmail());
+            User userByLogin = jdbcUserDao.findByLogin(user.getLogin());
             if (action.equals(ACTION_CREATE_USER)) {
                 //если логин не уникален
-                if (jdbcUserDao.findByLogin(user.getLogin()) != null) {
+                if (userByLogin != null) {
                     request.setAttribute("error_message", ERROR_NOT_UNIQUE_LOGIN);
                     return false;
                 }
@@ -199,6 +201,12 @@
                 jdbcUserDao.create(user);
                 request.setAttribute("message", MESSAGE_USER_CREATED);
             } else if (action.equals(ACTION_EDIT_USER)) {
+                String oldPassword = request.getParameter("oldPassword");
+                if (!userByLogin.getPassword().equals(oldPassword)) {
+                    request.setAttribute("error_message", ERROR_WRONG_PASSWORD);
+                    return false;
+                }
+
                 //если нашелся юзер с таким емейлом и это не текущий юзер
                 if (userByEmail != null) {
                     if (!userByEmail.getLogin().equals(user.getLogin())) {
Index: JSP/src/main/webapp/user_form.jsp
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- JSP/src/main/webapp/user_form.jsp	(date 1480521789000)
+++ JSP/src/main/webapp/user_form.jsp	(revision )
@@ -41,14 +41,14 @@
         </c:choose>
     </h3>
 
-    <c:if test="${error_message != null}">
-        <div class="message errorMessage" id="errorMessageLabel">${error_message}</div>
-    </c:if>
+
+     <div class="message errorMessage" id="errorMessageLabel">${error_message}</div>
+
     <c:if test="${message != null}">
         <div class="message">${message}</div>
     </c:if>
     <form action="admin" class="form-user" method="post" id="createUserForm"
-          onsubmit="return checkPasswordConfirm()">
+          onsubmit="return checkPasswordConfirm(${isCreate})">
         <input type="hidden" name="action"
                value="${action}">
         <div class="form-group row">
@@ -58,6 +58,17 @@
                        <c:if test="${!isCreate}">readonly="readonly"</c:if>  />
             </div>
         </div>
+
+        <c:if test="${!isCreate}">
+            <div class="form-group row">
+                <label class="col-xs-2 col-form-label">Old password</label>
+                <div class="col-xs-10">
+                    <input name="oldPassword" type="password" class="form-control" placeholder="old password" required
+                           id="oldPassword"/>
+                </div>
+            </div>
+        </c:if>
+
         <div class="form-group row">
             <label class="col-xs-2 col-form-label">Password</label>
             <div class="col-xs-10">
@@ -65,7 +76,6 @@
                        id="password"/>
             </div>
         </div>
-
         <div class="form-group row">
             <label class="col-xs-2 col-form-label">Confirm password</label>
             <div class="col-xs-10">
@@ -73,6 +83,7 @@
                        id="confirmPassword">
             </div>
         </div>
+
         <div class="form-group row">
             <label class="col-xs-2 col-form-label">Email</label>
             <div class="col-xs-10">
@@ -123,14 +134,21 @@
 
 <script>
 
-    function checkPasswordConfirm() {
+    function checkPasswordConfirm(isCreate) {
         var password = document.getElementById("password").value;
         var confirmPassword = document.getElementById("confirmPassword").value;
         var errorMessageLabel = document.getElementById("errorMessageLabel");
         if (password != confirmPassword) {
-            errorMessageLabel.innerHTML = "Error! Passwords do not match!";
+            errorMessageLabel.innerHTML = "Error! Password and confirm password do not match!";
             return false;
         } else {
+            if (!isCreate) {
+                var oldPassword = document.getElementById("oldPassword").value;
+                if (oldPassword == password) {
+                    errorMessageLabel.innerHTML = "Error! New password must be different from the old!";
+                    return false;
+                }
+            }
             return true;
         }
     }
